boot/freeldr/freeldr/ntldr/wlregistry.c

```diff
@@ -543,44 +560,42 @@ WinLdrScanRegistry(IN OUT PLIST_ENTRY BootDriverListHead,
     TRACE_CH(REACTOS, "BufferSize: %d \n", (int)BufferSize);
     TRACE_CH(REACTOS, "GroupNameBuffer: '%S' \n", GroupNameBuffer);
 
-    /* Loop through each group */
-    GroupName = GroupNameBuffer;
-    while (*GroupName)
-    {
-        TRACE("Driver group: '%S'\n", GroupName);
+    InitializeListHead(&DriverListHead);
 
-        /* Query the Order */
-        BufferSize = sizeof(OrderList);
-        rc = RegQueryValue(hOrderKey, GroupName, NULL, (PUCHAR)OrderList, &BufferSize);
-        if (rc != ERROR_SUCCESS) OrderList[0] = 0;
+    {
+        Index = 0;
 
-        /* enumerate all drivers */
-        for (TagIndex = 1; TagIndex <= OrderList[0]; TagIndex++)
+        while (TRUE)
         {
-            Index = 0;
+            /* Get the Driver's Name */
+            ValueSize = sizeof(ServiceName);
+            rc = RegEnumKey(hServiceKey, Index, ServiceName, &ValueSize, &hDriverKey);
+            TRACE("RegEnumKey(): rc %d\n", (int)rc);
 
-            while (TRUE)
+            /* Make sure it's valid, and check if we're done */
+            if (rc == ERROR_NO_MORE_ITEMS)
+                break;
+            if (rc != ERROR_SUCCESS)
             {
-                /* Get the Driver's Name */
-                ValueSize = sizeof(ServiceName);
-                rc = RegEnumKey(hServiceKey, Index, ServiceName, &ValueSize, &hDriverKey);
-                TRACE("RegEnumKey(): rc %d\n", (int)rc);
-
-                /* Make sure it's valid, and check if we're done */
-                if (rc == ERROR_NO_MORE_ITEMS)
-                    break;
-                if (rc != ERROR_SUCCESS)
-                {
-                    FrLdrHeapFree(GroupNameBuffer, TAG_WLDR_NAME);
-                    return;
+                FrLdrHeapFree(GroupNameBuffer, TAG_WLDR_NAME);
+
+                while (!IsListEmpty(&DriverListHead)) {
+                    DriverRegistryEntry* dre = CONTAINING_RECORD(RemoveHeadList(&DriverListHead), DriverRegistryEntry, ListEntry);
+
+                    FrLdrHeapFree(dre, TAG_WLDR_NAME);
                 }
-                //TRACE_CH(REACTOS, "Service %d: '%S'\n", (int)Index, ServiceName);
 
-                /* Read the Start Value */
-                ValueSize = sizeof(ULONG);
-                rc = RegQueryValue(hDriverKey, L"Start", &ValueType, (PUCHAR)&StartValue, &ValueSize);
-                if (rc != ERROR_SUCCESS) StartValue = (ULONG)-1;
-                //TRACE_CH(REACTOS, "  Start: %x  \n", (int)StartValue);
+                return;
+            }
+
+            /* Read the Start Value */
+            ValueSize = sizeof(ULONG);
+            rc = RegQueryValue(hDriverKey, L"Start", &ValueType, (PUCHAR)&StartValue, &ValueSize);
+            if (rc == ERROR_SUCCESS && StartValue == 0)
+            {
+                DriverRegistryEntry* dre;
```

Comment:
Coding style

Reply:
I've looked through https://reactos.org/wiki/Coding_Style and can't see anything relevant... what's the problem here?

Reply:
`STRUCT_NAME VariableName;` : that's the standard NT code style we use there.